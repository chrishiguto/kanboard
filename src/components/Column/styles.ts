import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;

  height: 100%;
`

export const Header = styled.div`
  ${({ theme }) => css`
    display: flex;

    margin-bottom: ${theme.spacings.xsmall};
  `}
`

export const Title = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.normal};
    text-transform: uppercase;
    color: #707990;
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    background: #e6eef5;
    flex: 1;

    padding: 0 ${theme.spacings.xsmall};
    border-radius: ${theme.border.radius};
  `}
`
