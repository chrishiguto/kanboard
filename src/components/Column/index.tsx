import { Droppable, DroppableProvided } from 'react-beautiful-dnd'

import * as S from './styles'

export type ColumnProps = {
  id: string
  children: React.ReactNode
  title?: string
}

const Column = ({ children, title, id }: ColumnProps) => (
  <S.Wrapper>
    {!!title && (
      <S.Header>
        <S.Title>{title}</S.Title>
      </S.Header>
    )}
    <Droppable droppableId={id}>
      {(provided: DroppableProvided) => (
        <S.Content ref={provided.innerRef} {...provided.droppableProps}>
          {children}
          {provided.placeholder}
        </S.Content>
      )}
    </Droppable>
  </S.Wrapper>
)

export default Column
