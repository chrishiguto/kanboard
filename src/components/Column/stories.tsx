import { Story, Meta } from '@storybook/react/types-6-0'
import Column from '.'

export default {
  title: 'Column',
  component: Column,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story = () => (
  <div style={{ maxWidth: 300, margin: '0 auto', height: 700 }}>
    <Column title="Title">Content</Column>
  </div>
)
