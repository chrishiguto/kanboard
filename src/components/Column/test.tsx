import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Column from '.'

describe('<Column />', () => {
  it('should render the column with title', () => {
    const { container } = renderWithTheme(
      <Column title="Column title">Column content</Column>
    )

    expect(
      screen.getByRole('heading', { name: /column title/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/column title/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the column without title', () => {
    renderWithTheme(<Column>Column content</Column>)

    expect(
      screen.queryByRole('heading', { name: /column title/i })
    ).not.toBeInTheDocument()
  })
})
