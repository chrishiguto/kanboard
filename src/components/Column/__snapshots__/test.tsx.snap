// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> should render the column with title 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.6rem;
}

.c2 {
  font-size: 1.4rem;
  font-weight: 400;
  text-transform: uppercase;
  color: #707990;
}

.c3 {
  background: #f6f8fa;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 1.6rem;
  border-radius: 0.8rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      Column title
    </h3>
  </div>
  <div
    class="c3"
  >
    Column content
  </div>
</div>
`;
